lighthouse-core/scripts/build-report-for-autodeployment.js

```diff
@@ -49,3 +51,68 @@ lhr.categories['lighthouse-plugin-someplugin'] = {
     }
   });
 })();
+
+/**
+ * Add a plugin to demo plugin rendering.
+ * @param {LH.Result} lhr
+ */
+function addPluginCategory(lhr) {
+  lhr.categories['lighthouse-plugin-someplugin'] = {
+    id: 'lighthouse-plugin-someplugin',
+    title: 'Plugin',
+    score: 0.5,
+    auditRefs: [],
+  };
+}
+
+/**
+ * Generate an LHR with errors for the renderer to display
+ * We'll write an "empty" artifacts file to disk, only to use it in auditMode
+ * @return {Promise<LH.Result>}
+ */
+async function generateErrorLHR() {
+  const url = 'http://fakeurl.com';
+  const options = {
+    requestedUrl: url,
+    settings: defaultSettings,
+    driver: {
+      getBrowserVersion: () => ({userAgent: 'Mozilla/5.0 ErrorUserAgent Chrome/66'}),
+    },
+  };
+  // @ts-ignore driver isn't mocked out completely
+  const artifacts = await GatherRunner.initializeBaseArtifacts(options);
+  // Add in a global runWarning
+  artifacts.LighthouseRunWarnings.push(`Something went wrong with recording the trace over your
+  page load. Please run Lighthouse again. (NO_FCP)`);
```

Comment:
> What do you think about checking in error artifacts? If the pageLoadError-defaultPass trace and devtoolsLog are deleted, they should be quite small.

could just replace all of the above with 
```js
/** @type {LH.BaseArtifacts} */
const baseArtifacts = {
  fetchTime: '2019-06-26T23:56:58.381Z',
  LighthouseRunWarnings: [
    `Something went wrong with recording the trace over your page load. Please run Lighthouse again. (NO_FCP)`, // eslint-disable-line max-len
  ],
  TestedAsMobileDevice: true,
  HostUserAgent: 'Mozilla/5.0 ErrorUserAgent Chrome/66',
  NetworkUserAgent: 'Mozilla/5.0 ErrorUserAgent Chrome/66',
  BenchmarkIndex: 1000,
  WebAppManifest: null,
  Stacks: [],
  settings: defaultSettings,
  URL: {
    requestedUrl: 'http://fakeurl.com',
    finalUrl: 'http://fakeurl.com',
  },
  Timing: [],
  PageLoadError: null,
  devtoolsLogs: {},
  traces: {},
};
```
and then continue as below. Only a little bit longer and the type assertion keeps everything in sync.